src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -660,7 +610,15 @@ private boolean shouldInstantiate(TypeElement site, Element e) {
                !((DeclaredType)e.getEnclosingElement().asType()).getTypeArguments().isEmpty();
     }
 
+    public record OverrideInfo(ExecutableElement overriddenMethod,
+                               DeclaredType overriddenMethodOwner) { }
+
     /*
+     * The record is used to pass the method along with the type where that method is visible.
+     * Passing the type explicitly allows to preserve a complete type information, including
+     * parameterization, which is otherwise unavailable without computation similar to what
+     * this method does.
+     *
```

Comment:
Consider moving this part of the comment onto the comment for the record declaration itself.